### PR TITLE
Template library

### PR DESCRIPTION
**Pull request by jiwalker-usgs**
Some major changes in dependency resolution.  Introduces templates that bring their dependencies with them. @wdwatkins footer is effected by this if you are looking into that currently.

### CODE REVIEW

**Comment by jiwalker-usgs** [comment]
@wdwatkins hold off on this for a second, I missed a dependency.  Also, let me pull in the latest changes and recheck before diving in on this.

**Comment by jiwalker-usgs** [comment]
@wdwatkins this is good to go now, I still haven't fixed the footer wonkiness though.

File: R/publish.R [hunk 1]
```diff
@@ -16,39 +16,40 @@ publish.character <- function(viz) {
...
-    names(dependencies) <- c(viz[['depends']], vizlabjs)
-  }
-  dependencies <- c(dependencies, recursive = TRUE)
+  # also manually put resources into context
+  context <- replaceOrAppend(template[['context']], viz[['context']])
+  context[['info']] <- replaceOrAppend(getBlocks("info", keep.block=F)[[1]], context[['info']])
 
-  context <- buildContext(viz, dependencies)
+  # flatten dependencies before lookups
+  dependencies <- c(dependencies, recursive = TRUE)
```
Comment (wdwatkins) [inline R/publish.R]: Didn't know about `recursive` 👍

Comment (wdwatkins) [inline R/publish.R]: Didn't know about `recursive` 👍

**Comment by wdwatkins** [comment]
Did Travis ever run on this?

**Pull request merged**